packages/plugins/plugin-assistant/src/components/Timeline/Timeline.tsx

```diff
@@ -68,163 +27,278 @@ export type Commit = {
-            if (parentSpan) {
-              parentSpan.end = Math.max(parentSpan.end, index);
+        const parents = commit.parents ?? [];
+        for (const parent of parents) {
+          const branch = getBranch(parent);
+          if (branch && branch !== commit.branch) {
+            span.start = Math.min(span.start, getCommitIndex(parent));
+
+            // Detect merge.
+            if (parents.length > 1) {
+              const parentSpan = spans.get(branch);
+              if (parentSpan) {
+                parentSpan.end = Math.max(parentSpan.end, index);
+              }
             }
           }
         }
-      }
-    });
+      });
 
-    return spans;
-  }, [commits, branches]);
+      return spans;
+    }, [commits, branches]);
 
-  const getPaths = (commit: Commit, branch: string, span: Span, index: number): string[] => {
+    // Navigation.
+    const containerRef = useRef<HTMLDivElement>(null);
+    const [current, setCurrent] = useState<number | undefined>();
+    const currentCommit = useMemo(() => (current !== undefined ? commits[current] : undefined), [current, commits]);
+    useEffect(() => {
+      onCurrentChange?.({ current, commit: current === undefined ? undefined : commits[current] });
+      const el = containerRef.current?.querySelector(`[data-index="${current}"]`);
+      el?.scrollIntoView({ behavior: 'instant', block: 'nearest' });
+    }, [current]);
+    useEffect(() => {
+      return addEventListener(containerRef.current!, 'keydown', (event) => {
+        switch (event.key) {
+          case 'ArrowDown': {
+            event.preventDefault(); // Prevent implicit scrolling.
+            if (event.metaKey) {
+              setCurrent(commits.length - 1);
+            } else {
+              setCurrent((selected) => (selected === undefined ? 0 : Math.min(commits.length - 1, selected + 1)));
+            }
+            break;
+          }
+          case 'ArrowUp': {
+            event.preventDefault(); // Prevent implicit scrolling.
+            if (event.metaKey) {
+              setCurrent(0);
+            } else {
+              setCurrent((selected) => (selected === undefined ? commits.length - 1 : Math.max(0, selected - 1)));
+            }
+            break;
+          }
+          case 'Escape': {
+            setCurrent(undefined);
+            break;
+          }
+        }
+      });
+    }, [commits.length, containerRef.current]);
+
+    return (
+      <ScrollContainer ref={scrollerRef}>
+        <div tabIndex={0} className={mx('flex flex-col is-full !outline-none', classNames)} ref={containerRef}>
+          {commits.map((commit, index) => {
+            // Skip branches that are not whitelisted.
+            const idx = getBranchIndex(commit.branch);
+            if (idx === -1) {
+              return null;
+            }
+
+            return (
+              <div
+                key={commit.id}
+                data-index={index}
+                aria-current={current === index}
```

Comment:
`aria-current` is for a master-detail pattern where the item that is displayed in the detail is marked with this attribute in the master.

If this is the idea of “active“, `aria-selected` is more appropriate, but it depends on the role, and has no role and isn’t focusable. What does selection do here?

If it’s just a visual style, Radix uses `data-selected` to avoid conflicts with a11y standards, you could do that here.

Reply:
At the time iirc we’d been discussing master-detail scenarios like NavTree and Mailbox, where `aria-current` is appropriate. I can’t tell from the screenshot or code exactly how this is meant to work, but it looks like a `listbox` for which `aria-selected` would be the appropriate attribute: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/listbox_role